common/djangoapps/student/helpers.py

```diff
@@ -198,6 +220,75 @@ def auth_pipeline_urls(auth_entry, redirect_url=None):
     }
 
 
+def _update_email_opt_in(request, org):
+    """Helper function used to hit the profile API if email opt-in is enabled."""
+
+    email_opt_in = request.POST.get('email_opt_in')
+    if email_opt_in is not None:
+        email_opt_in_boolean = email_opt_in == 'true'
+        preferences_api.update_email_opt_in(request.user, org, email_opt_in_boolean)
+
+
+def enroll(user, course_id, request, check_access):
```

Comment:
Sure @Salomari1987. At the lms level. But we're already using the enrollment api in three places:

 - Marketing site
 - Android app
 - iOS app

So it should be safe to use it instead of this refactoring.